### DISCUSSION TITLE
Migration from 2.3.5 to 3.latest

### DISCUSSION THREAD

**Discussion by kazikskrz** [hultenvp/solis-sensor#264] [Q&A] [answered]
Hi. I have 2 inverters running stable 2.3.5 version meaning my config exists in configuration.yaml (below). 
I have collected Energy Statistics for over 2 years and now it's time to migrate to latest version I guess.
Could you direct me how to properly do it without loosing collected statistics?
My current yaml setup:

# Sofar
  - platform: solis
    name: "Inverter"
    portal_domain: "m.ginlong.com"
    portal_username: !secret solis_username
    portal_password: !secret solis_password
    portal_plant_id: "****"
# Sofar
  - platform: solis
    name: "Sofar"
    portal_domain: "m.ginlong.com"
    portal_username: !secret solis_username
    portal_password: !secret solis_password
    portal_plant_id: "****"  

Among others it returns 2 main entities which I us for Energy Dashboard
![image](https://user-images.githubusercontent.com/54333498/220905388-13e72085-1157-4592-80d5-9dbffeaead10.png)

With latest version I have a problem:
While adding integration via UI and providing Inverter names: Solis, Sofar (for better identification) I'm loosing my statistics. 
Not directly but when I rename both sensors "...energy_today2" to be "...energy_today". 
Is there a proper way to approach it correctly?

**Comment by hultenvp** [comment, marked as answer]
When upgrading the config is automatically imported without any impact on your history.
Once successfully imported you'll get a notification to delete the entries from the yaml file

To be sure I'd propose the following sequence:
- Make a full backup
- Upgrade the solis integration to latest.
- Let it run for a while to confirm your stats are not messed up
- If you encounter issues:
    - Please save the logs and share them so I can help
    - Revert to backup.
- If all runs smoothly, delete entries from yaml, no need to restart, it's fine to wait till next time you need to restart anyhow.
